textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -72,25 +72,28 @@ public void open() throws DataFlowException {
             TopDocs topDocs = luceneIndexSearcher.search(dataReaderPredicate.getLuceneQuery(), Integer.MAX_VALUE);
             scoreDocs = topDocs.scoreDocs;
             cursor = OPENED;
+            
+            this.schema = dataReaderPredicate.getDataStore().getSchema();
 
-            this.queryTokens = Utils.tokenizeQuery(dataReaderPredicate.getLuceneAnalyzer(),dataReaderPredicate.getQueryString());
+            if (this.dataReaderPredicate.getIsSpanInformationAdded()) {

```

Comment:
I assume this `getIsSpanInformationAdded()` function gives enough info for the data reader for a regex matcher.